akka-distributed-data/src/main/scala/akka/cluster/ddata/ORMap.scala

```diff
@@ -383,12 +383,17 @@ final class ORMap[A, B <: ReplicatedData] private[akka] (
       case putOp: PutDeltaOp[A, B] ⇒
         val key = putOp.value._1
         thatValueDeltas += (key → (putOp.value :: Nil)) // put is destructive!
-      case _: RemoveDeltaOp[A, B] ⇒
-      // remove delta is only for the side effect of key being removed
-      // please note that if it is not preceded by update clearing the value
-      // anomalies will result
+      case removeOp: RemoveDeltaOp[A, B] ⇒
```

Comment:
Fix for #22648 but would be incorrect without corresponding fix in `ORSet` below